### DISCUSSION TITLE
Is it true that zustand store can only be object?

### DISCUSSION THREAD

**Discussion by samsze0** [pmndrs/zustand#1950] [Q&A]
As title. I try to make the zustand store an array (i.e. `StoreApi<SomeType[]>`) but in the store's subscribe callback, the state (supposedly an array) is somehow an object

I would like to mimic the way jotai works (atoms and derivable atoms) with zustand. And the reason why I am not using jotai is because I need to be able to modify the store from outside of react

**Comment by dai-shi** [comment]
Zustand supports both object and primitive values. However, arrays are a bit tricky as they are considered as objects. For now, I would suggest `replace=true` and to merge by yourself.

```js
const useArrayStore = create(() => [])
const pushItem = (item) => useArrayStore.setState((prev) => [...prev, item], true)
```


---

btw, Jotai supports updating atom values from outside, since v2.
